metagraph/modules/shared_data/src/main/scala/com/my/dor_metagraph/shared_data/combiners/DeviceCheckIn.scala

```diff
@@ -35,4 +31,63 @@ object DeviceCheckIn {
       CheckInDataCalculatedState(devices)
     )
   }
+
+  private def getRewardEpoch(epochProgress: EpochProgress): (Long, Long) = {
+    val currentEpoch: Long = epochProgress.value.value
+    val currentEpochModulus: Long = currentEpoch % EpochProgress1Day
+    val nextRewardEpoch: Long = currentEpoch - currentEpochModulus + EpochProgress1Day
+    (currentEpoch, nextRewardEpoch)
+  }
+
+  private def nextEpochProgressToReward(
+    epochProgress  : EpochProgress,
+    maybeDeviceInfo: Option[DeviceInfo]
+  ): Long = {
+    val (currentEpoch: Long, nextRewardEpoch: Long) = getRewardEpoch(epochProgress)
+
+    maybeDeviceInfo
+      .filter(_.nextEpochProgressToReward > currentEpoch)
+      .map(_.nextEpochProgressToReward)
+      .getOrElse(nextRewardEpoch)
+  }
+
+  private def getAnalyticsRewardsInformation(
+    epochProgress  : EpochProgress,
+    maybeDeviceInfo: Option[DeviceInfo],
+    dorAPIResponse : DorAPIResponse
+  ): Option[AnalyticsBountyInformation] =
+    dorAPIResponse.lastBillingId.map { lastBillingId =>
+      maybeDeviceInfo.fold(createAnalyticsBountyInformation(epochProgress, dorAPIResponse)) { deviceInfo =>
+        deviceInfo.analyticsBountyInformation.fold(createAnalyticsBountyInformation(epochProgress, dorAPIResponse)) { oldAnalyticsBountyInformation =>
+          updateAnalyticsBountyInformation(epochProgress, dorAPIResponse, lastBillingId, oldAnalyticsBountyInformation)
+        }
+      }
+    }
```

Comment:
This is a bit hard to read. Is this equivalent?
```suggestion
    dorAPIResponse
      .lastBillingId
      .map { lastBillingId =>
        maybeDeviceInfo
          .flatMap(_.analyticsBountyInformation)
          .map(updateAnalyticsBountyInformation(epochProgress, dorAPIResponse, lastBillingId, _))
          .getOrElse(createAnalyticsBountyInformation(epochProgress, dorAPIResponse))
      }
```

Reply:
Yeah, thanks